plugins/modules/win_acl_inheritance.ps1

```diff
@@ -1,75 +1,158 @@
-                If (-not $inheritedRule.IsInherited) {
-                    Continue
-                }
+function Remove-AclExplicitDuplicate {
+    [CmdletBinding(SupportsShouldProcess)]
+    [OutputType([System.Security.AccessControl.NativeObjectSecurity])]
+    param (
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [System.Security.AccessControl.NativeObjectSecurity] $acl
+    )
 
-                ForEach ($explicitRrule in $objACL.Access) {
-                    If ($explicitRrule.IsInherited) {
-                        Continue
-                    }
+    Begin {}
+
+    Process {
+        $properties = $acl.Access |
+            Get-Member -MemberType Property |
+            Where-Object { $_.Name -ne 'IsInherited' } |
+            Select-Object -ExpandProperty Name
 
-                    If (
-                        ($inheritedRule.FileSystemRights -eq $explicitRrule.FileSystemRights) -And
-                        ($inheritedRule.AccessControlType -eq $explicitRrule.AccessControlType) -And
-                        ($inheritedRule.IdentityReference -eq $explicitRrule.IdentityReference) -And
-                        ($inheritedRule.InheritanceFlags -eq $explicitRrule.InheritanceFlags) -And
-                        ($inheritedRule.PropagationFlags -eq $explicitRrule.PropagationFlags)
-                    ) {
-                        $objACL.RemoveAccessRule($explicitRrule)
+        ForEach ($inheritedRule in $($acl.Access | Where-Object { $_.IsInherited })) {
+            ForEach ($explicitRule in $($acl.Access | Where-Object { -not $_.IsInherited })) {
+                If ($null -eq (Compare-Object -ReferenceObject $explicitRule -DifferenceObject $inheritedRule -Property $properties)) {
+                    if ($PSCmdlet.ShouldProcess($acl)) {
+                        $acl.RemoveAccessRule($explicitRule)
                     }
                 }
             }
         }
+        return $acl
+    }
+
+    End {}
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$path = $module.Params.path
+if (-not $path.StartsWith('\\?\')) {
+    $path = [System.Environment]::ExpandEnvironmentVariables($path)
+}
+
+$reorganize = $module.Params.reorganize
+$state = $module.Params.state
+$check_mode = $module.CheckMode
+
+$module.Result.changed = $false
+
+$regeditHives = @{
+    'HKCR' = 'HKEY_CLASSES_ROOT'
+    'HKU' = 'HKEY_USERS'
+    'HKCC' = 'HKEY_CURRENT_CONFIG'
+}
 
-        Set-ACL -LiteralPath $path -AclObject $objACL -WhatIf:$check_mode
-        $result.changed = $true
+$pathQualifier = Split-Path -Path $path -Qualifier -ErrorAction SilentlyContinue
+$pathQualifier = $pathQualifier.Replace(':', '')
+
+if ($pathQualifier -in $regeditHives.Keys -and (-not (Test-Path -LiteralPath "${pathQualifier}:\"))) {
+    $null = New-PSDrive -Name $pathQualifier -PSProvider 'Registry' -Root $regeditHives.$pathQualifier
+    Push-Location -LiteralPath "${pathQualifier}:\"
```

Comment:
I seem to recall this is needed to do some `Set-Acl` on registry paths, is that actually true. While we do run code in Runspaces which shouldn't affect anything else I'm not the biggest fan of this. If we can use `$item.SetAccessControl($acl)` to set the info is this still needed?

If so you could probably do

```powershell
New-PSDrive -Name $pathQualifier -PSProvider 'Registry' -Root $regeditHives.$pathQualifier | Push-Location
```